react/GalleryLayout.tsx

```diff
@@ -99,17 +77,17 @@ const GalleryLayout: React.FC<GalleryLayoutProps> = ({
     ?.enableSearchRenderingOptimization
 
   return (
-    <ProductListProvider listName={trackingId}>
+    <ProductListProvider listName={trackingId as string}>
       <div className={galleryClasses}>
-        {rows.map((rowProducts, index) => (
+        {galleryRows.map((rowProducts, index) => (
           <GalleryLayoutRow
-            key={index.toString()}
+            key={`${currentLayoutOption.name}-${index}`}
```

Comment:
```suggestion
            key={currentLayoutOption.name}
```

Reply:
This removal makes the layout not to switch. It's not a good thing to set multiple elements with the same key, but if that was the case it should break when viewing any layout, not only when switching.

Also, the key changes when switching between layouts, so it shouldn't be a problem.
Maybe React uses these keys to unmount the elements, and since it finds multiple instances with the same keys, it doesn't know what to do and just leaves them there.

Reply:
Ops, yeah, totally! I understood this component wrong

I thought it was iterating through galleryLayoutOptions